lib/iris/fileformats/netcdf.py

```diff
@@ -1356,6 +1381,109 @@ def _create_cf_dimensions(
                     size = self._existing_dim[dim_name]
                 self._dataset.createDimension(dim_name, size)
 
+    def _add_mesh(self, cube):
+        """
+        Add the cube's mesh, and all related variables to the dataset.
+        Includes all the mesh-element coordinate and connectivity variables.
+
+        ..note::
+
+            Here, we do *not* add the relevant referencing attributes to the
+            data-variable, because we want to create the data-variable later.
+
+        Args:
+
+        * cube (:class:`iris.cube.Cube`):
+            A :class:`iris.cube.Cube` to be saved to a netCDF file.
+
+        Returns:
+            * cf_mesh_name (string or None):
+            The name of the mesh variable created, or None if the cube does not
+            have a mesh.
+
+        """
+        cf_mesh_name = None
+        mesh = cube.mesh
+        if mesh:
+            if mesh in self._name_coord_map.coords:
+                cf_mesh_name = self._name_coord_map.name(mesh)
+            else:
+                cf_mesh_name = self._create_mesh(cube, mesh)
+                self._name_coord_map.append(cf_mesh_name, mesh)
+
+            cf_mesh_var = self._dataset.variables[cf_mesh_name]
+
+            # Get the mesh-element dim names.
+            mesh_dims = self._mesh_dims[mesh]
+
+            # Add all the element coordinate variables.
+            for location in MESH_LOCATIONS:
+                coords_meshobj_attr = f"{location}_coords"
+                coords_file_attr = f"{location}_coordinates"
+                mesh_coords = getattr(mesh, coords_meshobj_attr, None)
+                if mesh_coords:
+                    coord_names = []
+                    for coord in mesh_coords:
+                        if coord is None:
+                            continue  # an awkward thing that mesh.coords does
+                        coord_name = self._create_generic_cf_array_var(
+                            cube,
+                            [],
+                            coord,
+                            element_dims=(mesh_dims[location],),
+                        )
+                        coord_names.append(coord_name)
+                    # Record the coordinates (if any) on the mesh variable.
+                    if coord_names:
+                        coord_names = " ".join(coord_names)
+                        _setncattr(cf_mesh_var, coords_file_attr, coord_names)
+
+            # Add all the connectivity variables.
+            # pre-fetch the set + ignore "None"s -- looks like a bug ?
```

Comment:
@pp-mo It's not a bug, it's returning a `namedtuple`, which will have populated slots, or a `None`.

Reply:
:+1: will amend